src/MeetingsSDKAdapter.js

```diff
@@ -942,31 +939,39 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
    *
    * @param {string} ID Meeting ID
    * @param {string} cameraID ID of the camera to switch to
-   * @private
    */
   async switchCamera(ID, cameraID) {
-    const sdkMeeting = this.fetchMeeting(ID);
-    const {stream, permission} = await this.getStream(
-      ID,
-      {sendVideo: true},
-      {video: {deviceId: cameraID}},
-    ).toPromise();
+    await this.updateMeeting(ID, async (meeting) => {
+      let updates;
+      const {stream, permission} = await this.getStream(
+        ID,
+        {sendVideo: true},
+        {video: {deviceId: cameraID}},
+      ).toPromise();
+
+      if (stream) {
+        updates = {
+          localVideo: {
+            stream,
+          },
+          cameraID,
+        };
 
-    if (stream) {
-      this.meetings[ID].localVideo.stream = stream;
-      this.meetings[ID].cameraID = cameraID;
+        if (meeting.state === MeetingState.JOINED) {
+          const sdkMeeting = this.fetchMeeting(ID);
 
-      if (this.meetings[ID].state === MeetingState.JOINED) {
-        await sdkMeeting.updateVideo({
-          stream,
-          receiveVideo: mediaSettings.receiveVideo,
-          sendVideo: mediaSettings.sendVideo,
-        });
+          await sdkMeeting.updateVideo({
+            stream,
+            receiveVideo: mediaSettings.receiveVideo,
+            sendVideo: mediaSettings.sendVideo,
+          });
+        }
+      } else {
+        throw new Error('Could not change camera, permission not granted:', permission);
```

Comment:
`Error` takes a single parameter, a message, unlike console.error which can take any number of params.

You have to build the complete error message yourself:
```js
throw new Error('Could [...] granted: ' + permission);
```